Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp

```diff
@@ -227,8 +227,15 @@ const struct wl_registry_listener registryListener = {
         else if (interfaceName == "xdg_wm_base"_s)
             priv->xdgWMBase = static_cast<struct xdg_wm_base*>(wl_registry_bind(registry, name, &xdg_wm_base_interface, 1));
         // FIXME: support zxdg_shell_v6?
-        else if (interfaceName == "wl_seat"_s)
+        else if (interfaceName == "wl_seat"_s) {
             priv->wlSeat = makeUnique<WPE::WaylandSeat>(static_cast<struct wl_seat*>(wl_registry_bind(registry, name, &wl_seat_interface, std::min<uint32_t>(version, 8))));
+            priv->wlSeat->setAvailableInputDevicesChangedCallback([weakDisplay = GWeakPtr { display }]() {
+                if (GRefPtr<WPEDisplayWayland> protectedDisplay { weakDisplay.get() }) {
```

Comment:
GWeakPtr is not thread safe and wpe-platform api either, so here we don't need to protect the display, we can just check the weak and get the pointer. So I would early return and then just use weakDisplay.get() directly.